tests/test_connector.rs

```diff
@@ -646,3 +658,218 @@ fn test_get_accounts_counter_and_transfer() {
+    let proof: Proof = serde_json::from_str(proof).unwrap();
+
+    let res = master_account.call(
+        CONTRACT_ACC.to_string(),
+        "deposit",
+        &proof.try_to_vec().unwrap(),
+        DEFAULT_GAS,
+        0,
+    );
+    let promise = &res.promise_results()[res.promise_results().len() - 2];
+    check_execution_status_failure(
+        promise.as_ref().unwrap().outcome().clone().status,
+        ERR_NOT_ENOUGH_BALANCE_FOR_FEE,
+    );
+}
+
+fn check_execution_status_failure(es: ExecutionStatus, err_msg: &str) {
+    match es {
+        ExecutionStatus::Failure(err) => {
+            assert!(err.to_string().contains(err_msg));
+        }
+        _ => panic!(),
+    }
+}
```

Comment:
This is a pretty funny way to check the error message, though should work :) I'm ok with introducing this as the other way needs to include some errors from crates and have many nested error types checks (and also has some linking issues at the moment).

Reply:
Btw, as the type of errors, we are checking there has the following format when converted to string:
`Action #0: Smart contract panicked: ERR_PROOF_EXIST [src/some_file.rs:LINE_NUMBER:COLUMN_NUMBER]`,
it makes sense to check that it contains not only the `err_msg` but `"Smart contract panicked: " + err_msg + " "`.
In this case, we are exhaustively guarded in case if we introduce errors that have the same matching prefix (e.g. `ERR_NOT_ENOUGH_BALANCE` and `ERR_NOT_ENOUGH_BALANCE_FOR_FEE`).

Reply:
So it's a guaranty that ExecutionOutcome contain exact error because:
1. We fetch ExecutionStatus::Failure - so it contains a failure message
2. We fetch specific messages from failure content.
So it's pretty enough.

Reply:
This works until we have no single error message that is the subset of any other.
I recommend adding this, though it's fine for me to keep it as is to save some time.

Reply:
If we have the same error messages for the same process flow we don't have any possibilities to extract that message.

Reply:
Ok, I'm fine if we don't add this right now. Though we are adding another thing to the technical debt :)